cartographer_ros/cartographer_ros/assets_writer.cc

```diff
@@ -54,6 +54,65 @@ namespace {
 constexpr char kTfStaticTopic[] = "/tf_static";
 namespace carto = ::cartographer;
 
+std::tuple<carto::mapping::proto::PoseGraph,
+           carto::mapping::proto::AllTrajectoryBuilderOptions>
+CreatePoseGraph(const std::string& pose_graph_filename,
+                const int num_bag_files) {
+  carto::mapping::proto::PoseGraph pose_graph_proto;
+  carto::mapping::proto::AllTrajectoryBuilderOptions
+      all_trajectory_builder_options;
+
+  carto::io::ProtoStreamReader reader(pose_graph_filename);
+  CHECK(reader.ReadProto(&pose_graph_proto));
+  CHECK(reader.ReadProto(&all_trajectory_builder_options));
+  CHECK_EQ(pose_graph_proto.trajectory_size(), num_bag_files)
```

Comment:
If this function takes the `num_bag_files` only for this `CHECK`: Move it to the call site instead?

Reply:
Done